coderd/apikey.go

```diff
@@ -133,16 +151,16 @@ func (api *API) postAPIKey(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusCreated, codersdk.GenerateAPIKeyResponse{Key: cookie.Value})
 }
 
-// @Summary Get API key
-// @ID get-api-key
+// @Summary Get API key by ID
+// @ID get-api-key-by-id
 // @Security CoderSessionToken
 // @Produce json
 // @Tags Users
 // @Param user path string true "User ID, name, or me"
 // @Param keyid path string true "Key ID" format(uuid)
 // @Success 200 {object} codersdk.APIKey
 // @Router /users/{user}/keys/{keyid} [get]
-func (api *API) apiKey(rw http.ResponseWriter, r *http.Request) {
+func (api *API) apiKeyByID(rw http.ResponseWriter, r *http.Request) {
```

Comment:
Should we just allow this by name or ID? And not make a new route?

We do this for users. We accept their id or username.

Reply:
I looked into this idea this morning because it's appealing, but I'm not sure it's practical in application:
 - Not all keys will have names, which may be confusing for the caller.
 - Because IDs and names are both type `string`, the implementation is a little more convoluted than it is for users. 
 - From a product perspective, we want folks deleting by name (not ID) from the CLI. So cobra cmd description for token removal might be a bit mismatched.

If you feel like I'm missing something, let me know.

Reply:
I figured the implementation would be checking both `ByID` and `ByName`. Which is unfortunate, but not terrible. It would be werid if someone named an api key the id of another :thinking: 

I can live with it being different.